UNFCCC_GHG_data/UNFCCC_reader/Guinea/read_GIN_BUR1_from_pdf.py

```diff
@@ -0,0 +1,905 @@
+    df_inventory_long["data"] = df_inventory_long["data"].str.replace("NE1", "NE")
+
+    # make sure all col headers are str
+    df_inventory_long.columns = df_inventory_long.columns.map(str)
+    df_inventory_long = df_inventory_long.drop(columns=["orig_cat_name"])
+
+    df_all_dict[page] = df_inventory_long
+
+# combine tables from all pages
+df_all = pd.concat([df_all_dict['110'], df_all_dict['111'], df_all_dict['112'], df_all_dict['113']],
+                   axis=0,
+                   join='outer').reset_index(drop=True)
+
+print("Converting to interchange format.")
+df_all_IF = pm2.pm2io.convert_long_dataframe_if(
+    df_all,
+    coords_cols=coords_cols,
+    coords_defaults=coords_defaults,
+    coords_terminologies=coords_terminologies,
+    coords_value_mapping=coords_value_mapping['main'],
+    filter_remove=filter_remove,
+    meta_data=meta_data,
+    convert_str=True,
+    time_format="%Y",
+)
+
+# There are inconsistent values in the main and the lulucf table
+# It looks like they put the values from 1990 again for 2019 in the main table.
+# The values from the lulucf table are assumed to be the correct ones.
+df_all_IF.loc[(df_all_IF["category (IPCC1996_2006_GIN_Inv)"] == "3") & (df_all_IF["entity"] == "CO") , "2019"] = 27.406
+df_all_IF.loc[(df_all_IF["category (IPCC1996_2006_GIN_Inv)"] == "3.C") & (df_all_IF["entity"] == "CO") , "2019"] = 27.406
+df_all_IF.loc[(df_all_IF["category (IPCC1996_2006_GIN_Inv)"] == "3.C.1") & (df_all_IF["entity"] == "CO") , "2019"] = 27.406
+
+# Values for category 3 and N2O are identical for 1990 and 2019
+# The sum of the sub-categories does not equal the value of the parent category
+# The value  in the lulucf table should therefore be the correct one
+df_all_IF.loc[(df_all_IF["category (IPCC1996_2006_GIN_Inv)"] == "3") & (df_all_IF["entity"] == "N2O") , "1990"] = 2.190
+
+# Values for category 3 and NOx are identical for 1990 and 2019
+# Replacing the duplicate value with the value from the lulucf table
+df_all_IF.loc[(df_all_IF["category (IPCC1996_2006_GIN_Inv)"] == "3") & (df_all_IF["entity"] == "NOx") , "2019"] = 1.644
+df_all_IF.loc[(df_all_IF["category (IPCC1996_2006_GIN_Inv)"] == "3.C") & (df_all_IF["entity"] == "NOx") , "2019"] = 1.644
+df_all_IF.loc[(df_all_IF["category (IPCC1996_2006_GIN_Inv)"] == "3.C.1") & (df_all_IF["entity"] == "NOx") , "2019"] = 1.644
+
+### convert to primap2 format ###
+data_pm2_main = pm2.pm2io.from_interchange_format(df_all_IF)
+
+# ###
+# 2. Read energy sector tables
+# ###
+
+pages = ['116', '117', '118', '119']
+df_energy_dict = {}
+for page in pages :
+    print("-" * 45)
+    print(f"Reading table from page {page}.")
+
+    tables_inventory_original = camelot.read_pdf(
+        str(input_folder / pdf_file),
+        pages=page,
+        flavor="lattice",
+        split_text=True
+    )
+
+    print("Reading complete.")
+
+    # cut last two lines of second table to ignore additional information regarding biomass for energy production
```

Comment:
Why not read the data? And if you want to ignore it it's better to use the filter in the conversion to IF to make ignoring explicit in the configuration (in case we later decide to read the data)

Reply:
> Why not read the data? And if you want to ignore it it's better to use the filter in the conversion to IF to make ignoring explicit in the configuration (in case we later decide to read the data)

I added `M.BIO`to the data set. 



> The following deletion can also be done via the `filter_remove` in the conversion to interchange format.

I would like to use `filter_remove` in `convert_long_dataframe_if`, but I'm not sure how. The version of the data frame that goes into the function has different category names, because I applied the regular expressions before. For example, I would like to add `Information Items` to the categories that should be removed, but at this point it has been transformed to `Information`. So I guess `filter_remove` wouldn't catch it. To me it seems to delete the unwanted categories earlier, but maybe I'm missing something.